odoo/tools/mail.py

```diff
@@ -260,6 +260,9 @@ def html_sanitize(src, silent=True, sanitize_tags=True, sanitize_attributes=Fals
 TEXT_URL_REGEX = r'https?://[a-zA-Z0-9@:%._\+~#=/-]+(?:\?\S+)?'
 # retrieve inner content of the link
 HTML_TAG_URL_REGEX = URL_REGEX + r'([^<>]*>([^<>]+)<\/)?'
+HTML2TEXT_NEWLINES = re.compile('<(div|p|br|tr)[^>]*>|\n')
+# From c24b @ https://stackoverflow.com/a/12982689/18408071 - CC BY-SA 4.0
+HTML2TEXT_STRIP_HTML = re.compile('<.*?>|&([a-z0-9]+|#[0-9]{1,6}|#x[0-9a-f]{1,6});')
```

Comment:
I feel like we might have some corner cases where the HTML are not stripped, as you do it after the HTML sanitation, you can just do something like
```
HTML2TEXT_STRIP_HTML = re.compile('<.*?>')
```

(because all `<>` that does not belong to an HTML tag will be escaped)

Reply:
Ah ok I see, for which use case the second part is used ?
```
|&([a-z0-9]+|#[0-9]{1,6}|#x[0-9a-f]{1,6});
```

Reply:
> it also covers characters that are not in <>, such as **`&lt;`**, which we don't want to see either. See test_html2text_only.

If you have a single `<` for example, sanitation will convert it to `&lt;`, which, while not a real problem in an email preview, it's not exactly 'text only" either :thinking:

I'll add this as a parameter, if you'll allow this one, because otherwise function name is misleading... and I don't think we should rename it to be only a `html2_email_preview`, right?

Reply:
Mhh, isn't it better to just use the regex `<.*?>` and then html unescape the result ?

If you have some special character in the text they will be removed